minio/docs
# MinIO Kubernetes Skeleton (In Progress)

NOT READY FOR MERGE

To test locally:

- Check out the repository + branch
- Run `pip install -r requirements.txt` (create a virtual environment first if you want)
- Run `make html` from repo root. Build outputs to `build/html`. Open `index.html` in browser.

First Pass (Ready For Review)

- `kubectl-minio` reference
- Direct CSI reference

ToDo:

- `kubectl-minio` Quick Reference
- `kubectl-minio` Common Operations
- MinIO `operator` CRD Reference
- MinIO `operator` `kustomize` Guidance
- Deploy MinIO Tenant (`kubectl-minio`)
- Expand MinIO Tenant (`kubectl-minio`)
- MinIO KES Reference + Deployment
- MinIO Console Reference + Deployment

Thoughts:

- KES / Console Reference might be best left under the operator CRD reference for simplicity. 
- Since both KES / Console are tightly woven into deploying a tenant, might need to integrate those into the standard deployment tutorials as an *always do this* sort of thing
- Need to figure out eventually how to deploy/expand KES/Console for an existing setup.
